stan/math/rev/core/profiling.hpp

```diff
@@ -115,7 +116,24 @@ class profile_info {
 
 using profile_key = std::pair<std::string, std::thread::id>;
 
-using profile_map = std::map<profile_key, profile_info>;
+namespace internal {
+struct hash_profile_key {
+  std::size_t operator()(const profile_key& key) const {
+    return std::hash<std::string>()(key.first)
+           ^ std::hash<std::thread::id>()(key.second);
```

Comment:
Just want to double-check that this is intentional, since the placement of the `^` makes it look like a compiler error and suggestion 😅

Reply:
Lol yes intended